pkg/build/strategies/sti/sti.go

```diff
@@ -196,6 +205,15 @@ func (builder *STI) Prepare(config *api.Config) error {
 		WorkingDir: config.WorkingDir,
 	}
 
+	if len(config.RuntimeImage) > 0 {
+		// For pulling RuntimeImage currently we're using the same PullPolicy as for BuilderImage
+		_, err := dockerpkg.PullImage(config.RuntimeImage, builder.docker, config.BuilderPullPolicy, false)
```

Comment:
1. why we need pull this image at prepare function , is it possible pull this image after the first build complete ? If the first build failed, this pull is really no need.
2. Or if you want to pull this image at the prepare phase , is it possible for you to make this pull in a new gorouting in a asynchronous way ?